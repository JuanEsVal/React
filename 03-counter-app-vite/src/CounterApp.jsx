import { PropTypes } from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {

  const [counter, setCounter] = useState(value);

  const handleMinus = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);
  const handleAdd = () => setCounter((c) => c + 1);  //Toma el valor actual d counter

  console.log(counter);
  return (
    <>
      <h1> CounterApp </h1>
      <h2> {counter} </h2>
      <button onClick={handleMinus}> -1 </button>
      <button onClick={handleReset}> Reset </button>
      <button onClick={handleAdd}> +1 </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  value: 0,
};
