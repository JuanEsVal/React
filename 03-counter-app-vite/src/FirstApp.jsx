import { PropTypes } from 'prop-types'

// Realmente:
// h1 deberia estar dentro de un heder
// h2 deberia estar dentro deun parrafo al igual que h3... etc
 

export const FirstApp = ({title, subtitle, name}) => {
  return (
    <>
        <h1> {title} </h1>     
        <h2>{subtitle}</h2>
        <h3>{name}</h3>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  // subtitle: PropTypes.number.isRequired
}

// Valores por defecto si no llega informaci√≥n
FirstApp.defaultProps = {
  title: 'No hay Titulo',
  subtitle: 'No hay Subtitulo',
  name: 'Juan Valencia'
}