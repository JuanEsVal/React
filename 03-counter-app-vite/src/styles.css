html, body{
    background: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 10px;
}

button{
    padding: 10px;
    font-size: 1.2rem;
}