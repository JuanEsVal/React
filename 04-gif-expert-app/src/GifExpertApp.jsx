//Despues de la version 17 de React ya no necesitamos importar React from 'react
// import React from 'react'



export const GifExpertApp = () => {
  return (
    <>
      <h1>GitExpertApp</h1>
    </>
  )
}