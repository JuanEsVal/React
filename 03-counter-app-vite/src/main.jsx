// import './styles.css'
// import { FirstApp } from './FirstApp'

import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value = {100} />
  </React.StrictMode>
)



{/* <FirstApp title={'Hola Soy Goku'}/> */}
// import {App} from './App'
{/* <App title = {'Juan'} subtitle = {1980} /> */}