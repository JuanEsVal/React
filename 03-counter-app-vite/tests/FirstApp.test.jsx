import { render } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'

describe( 'Prueba en <FirstApp />', () => {

    test('Debe hacer Match con el Snapshot', () => { 

        const title = 'Hola, soy Juan Esteban'
        const { container } = render( <FirstApp title={title} />)

        // console.log(container);

        expect ( container ).toMatchSnapshot();
    })

  
    // test('Debe mostrar el titulo en un h1', () => { 

    //     const title = 'Hola, soy Esteban'
    //     const { container } = render( <FirstApp title={title} />)
    //     expect ( getByText(title) ).toBeTruthy();

    //     const h1 = container.querySelector('h1');
    //     expect ( h1.innerHTML ).toContain(title);

    // })

    test('Debe mostrar el subtitulo enviado por props', () => { 

        const title = 'Hola, soy Esteban'
        const subtitle = 'De Apellido Valencia'

        const { getAllByText } = render(<FirstApp title={title} subtitle={subtitle} />)
        expect ( getAllByText(subtitle).length).toBe(1);
    })

})