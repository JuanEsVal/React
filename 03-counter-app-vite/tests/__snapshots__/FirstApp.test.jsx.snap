// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <FirstApp /> Debe hacer Match con el Snapshot 1`] = `
<div>
  <h1>
     
    Hola, soy Juan Esteban
     
  </h1>
  <h2>
    No hay Subtitulo
  </h2>
  <h3>
    Juan Valencia
  </h3>
</div>
`;
